akka-persistence/src/main/mima-filters/2.5.4.backwards.excludes

```diff
@@ -0,0 +1,14 @@
+# LevelDB journal compaction #21677
+ProblemFilters.exclude[ReversedMissingMethodProblem]("akka.persistence.journal.leveldb.LeveldbStore.compactionIntervals")
+ProblemFilters.exclude[ReversedMissingMethodProblem]("akka.persistence.journal.leveldb.LeveldbStore.akka$persistence$journal$leveldb$LeveldbStore$_setter_$compactionIntervals_=")
+ProblemFilters.exclude[InheritedNewAbstractMethodProblem]("akka.persistence.journal.leveldb.CompactionSegmentManagement.compactionLimit")
+ProblemFilters.exclude[InheritedNewAbstractMethodProblem]("akka.persistence.journal.leveldb.CompactionSegmentManagement.compactionSegment")
+ProblemFilters.exclude[InheritedNewAbstractMethodProblem]("akka.persistence.journal.leveldb.CompactionSegmentManagement.akka$persistence$journal$leveldb$CompactionSegmentManagement$$latestCompactionSegments")
+ProblemFilters.exclude[InheritedNewAbstractMethodProblem]("akka.persistence.journal.leveldb.CompactionSegmentManagement.compactionIntervals")
+ProblemFilters.exclude[InheritedNewAbstractMethodProblem]("akka.persistence.journal.leveldb.CompactionSegmentManagement.updateCompactionSegment")
+ProblemFilters.exclude[InheritedNewAbstractMethodProblem]("akka.persistence.journal.leveldb.CompactionSegmentManagement.akka$persistence$journal$leveldb$CompactionSegmentManagement$$latestCompactionSegments_=")
+ProblemFilters.exclude[InheritedNewAbstractMethodProblem]("akka.persistence.journal.leveldb.CompactionSegmentManagement.mustCompact")
```

Comment:
isn't `CompactionSegmentManagement` a new class? then it should not be needed here

Reply:
Unfortunately @patriknw this (along with the *ActorLogging* issue bellow) is a more complex issue than it seems, and I will absolutely need your feedback on how to deal with it. These filters are mandated by the ```mimaReportBinaryIssues``` SBT task, otherwise binary compatibility errors are raised. The *InheritedNewAbstractMethodProblem* is raised due to the fact that *LeveldbStore* now mixes in the *CompactionSegmentManagement* and *ActorLogging*, thus inheriting the corresponding members from them. The thing is however that there seems to be a problematic approach in the implementation of the mima tool regarding this issue (had to download the mima sources to figure it out): If you check the [problems implementation](https://github.com/typesafehub/migration-manager/blob/master/core/src/main/scala/com/typesafe/tools/mima/core/Problems.scala) you will notice that the *InheritedNewAbstractMethodProblem* does include the class info of the class that inherits an abstract member, however that class info is only used to compose the **description** of the problem, and is not used to somehow override the **matchName**, which is the property used for matching by the [ProblemFilters](https://github.com/typesafehub/migration-manager/blob/master/core/src/main/scala/com/typesafe/tools/mima/core/Filters.scala) predicate. The implementation of **matchName** inherited via *AbstractMethodProblem* -> *MethodProblem* only uses the info of the method which introduces the problem, whose fully qualified name in the case of an inherited method is based on the original class/trait that introduces it (hence the misinterpretation one may end up to, thinking that the filter is about the class/trait that introduces the method, instead of the one inheriting it). 
The problem with the above is that, event though these filters actually do the job in this case, they are in fact **way too broad**. The fact that the filters fail to take the inheriting class info into account can lead to ignoring _ActorLogging_ mix-ins from any class/trait into the akka-persistence module, and that would result into false-negatives into binary compatibility checks. On the other hand not adding these filters produces binary compatibility errors which will lead to failure of the ```validatePullRequest``` SBT task. Thus the questing is, how do we proceed?

Reply:
I see. That is strange and slightly dangerous because after this it would not detect new breakages if ActorLogging was added to something else. I have created issue https://github.com/typesafehub/migration-manager/issues/196

Until that has been fixed we have to include these filters.